view-creator-framework/src/test/java/org/hypertrace/core/viewcreator/pinot/PinotUtilsTest.java

```diff
@@ -31,32 +33,52 @@ public void testCreatePinotSchemaForView() {
+    Assertions
+        .assertEquals(DataType.STRING, pinotSchemaForView.getDimensionSpec("name").getDataType());
+    Assertions
+        .assertEquals(DataType.BYTES, pinotSchemaForView.getDimensionSpec("id_sha").getDataType());
     Assertions.assertEquals(64, pinotSchemaForView.getDimensionSpec("id_sha").getMaxLength());
     Assertions.assertFalse(pinotSchemaForView.getDimensionSpec("friends").isSingleValueField());
-    Assertions.assertFalse(pinotSchemaForView.getDimensionSpec("properties__KEYS").isSingleValueField());
-    Assertions.assertEquals(DataType.STRING, pinotSchemaForView.getDimensionSpec("properties__KEYS").getDataType());
+    Assertions
+        .assertFalse(pinotSchemaForView.getDimensionSpec("properties__KEYS").isSingleValueField());
+    Assertions.assertEquals(DataType.STRING,
+        pinotSchemaForView.getDimensionSpec("properties__KEYS").getDataType());
     Assertions.assertEquals("",
         pinotSchemaForView.getDimensionSpec("properties__KEYS").getDefaultNullValue());
-    Assertions.assertFalse(pinotSchemaForView.getDimensionSpec("properties__VALUES").isSingleValueField());
+    Assertions.assertFalse(
+        pinotSchemaForView.getDimensionSpec("properties__VALUES").isSingleValueField());
     Assertions.assertEquals(DataType.STRING,
         pinotSchemaForView.getDimensionSpec("properties__VALUES").getDataType());
     Assertions.assertEquals("",
         pinotSchemaForView.getDimensionSpec("properties__KEYS").getDefaultNullValue());
-    Assertions.assertEquals(DEFAULT_DIMENSION_NULL_VALUE_OF_STRING, pinotSchemaForView.getDimensionSpec("name")
-                                            .getDefaultNullValue());
-    Assertions.assertEquals(DEFAULT_DIMENSION_NULL_VALUE_OF_STRING, pinotSchemaForView.getDimensionSpec("friends")
-                                            .getDefaultNullValue());
+    Assertions.assertEquals(DEFAULT_DIMENSION_NULL_VALUE_OF_STRING,
+        pinotSchemaForView.getDimensionSpec("name")
+            .getDefaultNullValue());
+    Assertions.assertEquals(DEFAULT_DIMENSION_NULL_VALUE_OF_STRING,
+        pinotSchemaForView.getDimensionSpec("friends")
+            .getDefaultNullValue());
     // metric fields are not part of dimension columns
-    Assertions.assertEquals("time_taken_millis", pinotSchemaForView.getMetricFieldSpecs().get(0).getName());
-    Assertions.assertEquals(DataType.LONG, pinotSchemaForView.getMetricFieldSpecs().get(0).getDataType());
+    Assertions.assertEquals("time_taken_millis",
+        pinotSchemaForView.getMetricFieldSpecs().get(0).getName());
+    Assertions
+        .assertEquals(DataType.LONG, pinotSchemaForView.getMetricFieldSpecs().get(0).getDataType());
+
+    Assertions.assertEquals(3, pinotSchemaForView.getDateTimeFieldSpecs().size());
+
+    DateTimeFieldSpec dateTimeFieldSpec = pinotSchemaForView
+        .getDateTimeSpec("creation_time_millis");
+    Assertions.assertEquals("creation_time_millis", dateTimeFieldSpec.getName());
```

Comment:
Shouldn't this be `end_time_millis` that is what was added in `sample-view-generation-spec.conf`?

Reply:
And `creation_time_millis` still should be `LONG` only, right? As it is still part of dimensions columns.

Reply:
Ah! I see, we do support `timeColumn` too.